miden-vm/src/cli/run.rs

```diff
@@ -166,12 +170,26 @@ fn run_masm_program(params: &RunCmd) -> Result<(ExecutionTrace, [u8; 32]), Repor
         }
     }
 
+    // validate kernel file if provided
+    if let Some(kernel_path) = &params.kernel_file {
```

Comment:
We should support specifying a kernel in either `.masp` (i.e. `Package`) or `.masm` forms. More often than not, the kernel will already be compiled to a package, so specifying the `.masp` will be more useful.

The same is true of any libraries passed in - I would expect most of them to be already assembled to packages.